staging/src/k8s.io/legacy-cloud-providers/azure/azure_standard.go

```diff
@@ -908,3 +1021,16 @@ func generateStorageAccountName(accountNamePrefix string) string {
 	}
 	return accountName
 }
+
+// GetNodeNameByIPConfigurationID gets the node name by IP configuration ID.
+func (as *availabilitySet) GetNodeNameByIPConfigurationID(ipConfigurationID string) (string, error) {
+	matches := nicIDRE.FindStringSubmatch(ipConfigurationID)
+	if len(matches) != 3 {
```

Comment:
I have test data from 1.21.0-alpha clusters built by aks-engine that suggests this PR broke ELB creation for clusters running Windows availability sets. AKS Engine creates Windows NIC interfaces with a name like `1598k8s01nic-1`. The regex in the `nicIDRE` reference above expects an explicit `-nic-` suffix (note the `-` before `nic-1`).

See this go playground example that proves that the above regex fails the match length test:

https://play.golang.org/p/Eh63N6EAOsc

It will be hard to change the name of the NIC that Windows VMs are attached in the AKS Engine ARM template logic without breaking backwards compatibility. Would it be possible to change the NIC regex to:

`/subscriptions/(?:.*)/resourceGroups/(?:.+)/providers/Microsoft.Network/networkInterfaces/(.+)nic-(.+)/ipConfigurations/(?:.*)`

cc @marosset @jsturtevant @nilo19 @jadarsie @CecileRobertMichon @devigned

This took longer than normal to verify it should have to discover due to the fact that we don't have a lot of AvailabilitySet tests for Windows.

@chewong how are we testing Windows + Availability Set nodes upstream? We should probably target catching these symptoms earlier in the upstream tests themselves.

Reply:
> how are we testing Windows + Availability Set nodes upstream?

We have https://testgrid.k8s.io/provider-azure-windows#pr-aks-engine-azure-windows and https://testgrid.k8s.io/provider-azure-windows#pr-aks-engine-azure-windows-containerd but I don't think we have a job that tests the scenario above.

Reply:
All aks-engine test passes on https://testgrid.k8s.io/sig-windows-azure use availability sets.

Reply:
I suspect this PR will break LB scenarios for all CAPZ VMs (not just Windows), as CAPZ naming for network interfaces is  `<machinename>-nic` (no `-` after "nic").

Reply:
We changed the logic of this function in another PR https://github.com/kubernetes/kubernetes/pull/97635

Reply:
FYI @nader-ziada